src/Raven.Server/Documents/Sharding/ShardedDocumentsStorage.cs

```diff
@@ -363,6 +367,10 @@ public ShardedDocumentDatabase.DeleteBucketCommand.DeleteBucketResult DeleteBuck
             // delete revisions for document
             RevisionsStorage.ForceDeleteAllRevisionsFor(context, document.Id);
             deleted++;
+
+            if (context.Transaction.InnerTransaction.LowLevelTransaction.NumberOfModifiedPages +
+                context.Transaction.InnerTransaction.LowLevelTransaction.AdditionalMemoryUsageSize.GetValue(SizeUnit.Bytes) / Constants.Storage.PageSize > MaxTransactionSize)
```

Comment:
can we move the size calculation to the `LowLevelTransaction`

Reply:
done